### PR TITLE
Fix target hash stability bug due to order of descendants

### PR DESCRIPTION
**Pull request by itamaro**

### CODE REVIEW

File: yabt/buildcontext.py [hunk 1]
```diff
@@ -128,11 +128,11 @@ def walk_target_deps_topological_order(self, target: Target):
 
     def generate_direct_deps(self, target: Target):
         """Generate only direct dependencies of `target`."""
-        yield from (self.targets[dep_name] for dep_name in target.deps)
+        yield from (self.targets[dep_name] for dep_name in sorted(target.deps))
```
Comment (eyalr-resonai) [inline yabt/buildcontext.py]: this does not feel to me the correct place to sort?

Comment (eyalr-resonai) [reply yabt/buildcontext.py]: oh sorrt, I take it back

Comment (itamaro) [reply yabt/buildcontext.py]: why not?

Comment (itamaro) [reply yabt/buildcontext.py]: > oh sorry, I take it back

oh, ok :-)

**Pull request merged**